src/Widgets/Views/MainView.vala

```diff
@@ -90,11 +94,11 @@ public class Bluetooth.Widgets.MainView : Gtk.Box {
         });
 
         object_manager.device_removed.connect ((device) => {
-            devices_box.get_children ().foreach ((child) => {
-                if (child is Bluetooth.Widgets.Device) {
-                    ((Bluetooth.Widgets.Device) child).destroy ();
-                }
-            });
+            if (devices.has_key (device.modalias)) {
```

Comment:
If you're still going for the HashMap, this could be optimised from two iterations on the `devices` HashMap to one without using `has_key`:

```
var widget = devices[device.modalias]
if (widget != null) {
    widget.no_show_all = true;
    widget.visible = false;
}

```